### ISSUE TITLE
Lavaan with a purposefully under-identified multiple group model 

### ISSUE THREAD

**Issue by YiFengEDMS** [yrosseel/lavaan#109]
Hello,

I was running an under-identified multiple group model using lavaan, and found that lavaan could output the results normally without a warning message. I originally have posted my question on lavaan Google group and Dr. Rosseel has kindly explained how lavaan worked it out in such a scenario.  

My lavaan code is as below:

```
mod.mg1='
group:1

int=~ 1*V1 + 1*V2
slp =~ 0*V1 + 1*V2
int ~~ start(0.292)*int+a*int  #variance
slp ~~ start(0.5)*slp + b*slp
int ~~ z*slp  #covariance
V1 ~~ r1*V1
V2 ~~ r2*V2   # residual variance for t2
int ~ c*1  # intercepts
slp ~ d*1
V1 ~ 0*1
V2 ~ 0*1

group:2
int =~ 1*V1 + 1*V2
slp =~ 0*V1 + 2*V2
int ~~ start(0.292)*int+a*int  #variance
slp ~~ start(0.5)*slp+b*slp
int ~~ z*slp  #covariance
V1 ~~ r1*V1
V2 ~~ r3*V2   # residual variance for t3
int ~ c*1  # intercepts
slp ~ d*1
V1 ~ 0*1
V2 ~ 0*1
'

fit.mg1=lavaan(mod.mg1,cdata.2,group="g",estimator="ML",information="observed",control=list(trace=0,init_nelder_mead=TRUE),mimic="Mplus",verbose=T)
```

I have also uploaded my simulated data as well as the Mplus code in my original post here:  https://groups.google.com/forum/#!topic/lavaan/LY0bd8XZIug 

Again, thank you so much for all your help and insights!

Best,
Yi

**Comment by YiFengEDMS** [comment]
Just a side note in case this could be of any help, in terms of detecting under-identified models, I find the "decomposition table" approach proposed in Hancock et al. (2009) very helpful, in both single-group and multiple-group scenarios:

Hancock, G. R., Stapleton, L. M., & Arnold-Berkovits, I. (2009). The tenuousness of invariance tests within multisample covariance and mean structure models. In T. Teo & M. S. Khine (Eds.), Structural equation modeling: Concepts and applications in educational research (pp. 137– 174). Rotterdam, the Netherlands: Sense.

**Comment by TDJorgensen** [comment]
When you post R syntax that take up multiple lines
``` 
like
this
``` 
, you need to use three backticks ( \``` ) above and below the syntax, instead of the single pair of backticks to format syntax inline (`like this`).  Currently, your asterisks are being interpreted as formatting for plain text (text between a single pair of asterisks will be italicized; between a double-pair of asterisks will be bold).

**Comment by YiFengEDMS** [comment]
Hi Dr. Jorgensen, thanks for pointing it out! I have changed the formatting accordingly.

**Comment by yrosseel** [comment]
I created a 'check' for this, that results in a warning() if lavaan thinks the variance-covariance matrix of the free parameters in the presence of equality constraints is not strictly positive definite. Available in dev 0.6-2.1277 or higher. It gives a warning for your model/dataset combination, but more testing is needed to see if this doesn't produce 'false alarms'. The test (and warning) can be shut off if you add the argument check.vcov = FALSE when calling sem()/lavaan().

**Comment by YiFengEDMS** [comment]
Thanks Dr. Rosseel! I just tried the code again in lavaan and I saw the warning message come up.
